bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/codemining/CodeMiningLineHeaderAnnotation.java

```diff
@@ -94,7 +98,17 @@ private int getMultilineHeight() {
 		}
 		numLinesOfAllMinings++;
 		StyledText styledText= super.getTextWidget();
-		return numLinesOfAllMinings * (styledText.getLineHeight() + styledText.getLineSpacing());
+		int lineHeight;
+		if (gc != null) {
+			// styledText.getLineHeight() returns 14 pixels for font "Consolas" size 9 on windows
+			// gc.textExtent() for italic character "f" returns 15 pixels
+			Point ext= gc.textExtent("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"); //$NON-NLS-1$
```

Comment:
can't we calculate the height based on the actual code mining text?

Reply:
done with 1a63102